velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -1217,7 +1232,8 @@ using OrderedPartition = std::pair<MaterializedRow, MaterializedRowMultiset>;
 // Returns true if equal.
 static bool compareOrderedPartitions(
     const OrderedPartition& expected,
-    const OrderedPartition& actual) {
+    const OrderedPartition& actual,
+    const RowTypePtr& valueType) {
```

Comment:
Why we need pass valueType?

Reply:
@xiaoxmeng This valueType is the Velox type of `expected.second`. findFloatingPointColumns() needs this to identify columns that contain floating-point types. If we don't pass this valueType,  findFloatingPointColumns() will have to infer the type from `expected.second` that is a `std::multiset<std::vector<velox::variant>>` (each std::vector<velox::variant> represents one row of a table). This inference is complex because we may not be able to infer the type by looking at only one std::vector<variant>. For example, if the variant is an array type, we can only infer the array's element type if the variant in the current std::vector happens to be a non-empty array. In the worse case, we need to loop over all std::vector<variant> in the std::multiset to determine the element type. 

I originally wrote this logic, but I later found this inference is unnecessary because we can get the exact Velox type.

Reply:
Thanks for the explanation. Could you just comment the function that we need the result data type to help identify the floating pointer column for comparison with epsilon? thanks!